crates/fuel-core/src/state/in_memory/transaction.rs

```diff
@@ -1,708 +0,0 @@
-where
-    Description: DatabaseDescription,
-{
-    fn iter_all(
-        &self,
-        column: Self::Column,
-        prefix: Option<&[u8]>,
-        start: Option<&[u8]>,
-        direction: IterDirection,
-    ) -> BoxedIter<KVItem> {
-        // iterate over inmemory + db while also filtering deleted entries
-        self.view_layer
-                // iter_all returns items in sorted order
-                .iter_all(column, prefix, start, direction)
-                // Merge two sorted iterators (our current view overlay + backing data source)
-                .merge_join_by(
-                    self.data_source.iter_all(column, prefix, start, direction),
-                    move |i, j| {
-                        if let (Ok(i), Ok(j)) = (i, j) {
-                            if IterDirection::Forward == direction {
-                                i.0.cmp(&j.0)
-                            } else {
-                                j.0.cmp(&i.0)
-                            }
-                        } else {
-                            // prioritize errors from db result first
-                            if j.is_err() {
-                                Ordering::Greater
-                            } else {
-                                Ordering::Less
-                            }
-                        }
-                    },
-                )
-                .map(|either_both| {
-                    match either_both {
-                        // in the case of overlap, choose the left-side (our view overlay)
-                        EitherOrBoth::Both(v, _)
-                        | EitherOrBoth::Left(v)
-                        | EitherOrBoth::Right(v) => v,
-                    }
-                })
-                // filter entries which have been deleted over the course of this transaction
-                .filter(move |item| {
-                    if let Ok((key, _)) = item {
-                        !matches!(
-                            self.changes[column.as_usize()]
-                                .lock()
-                                .expect("poisoned")
-                                .get(key),
-                            Some(WriteOperation::Remove)
-                        )
-                    } else {
-                        // ensure errors are propagated
-                        true
-                    }
-                }).into_boxed()
-    }
-}
-
-impl<Description> BatchOperations for MemoryTransactionView<Description> where
-    Description: DatabaseDescription
-{
-}
-
-impl<Description> TransactableStorage for MemoryTransactionView<Description>
-where
-    Description: DatabaseDescription,
-{
-    fn flush(&self) -> DatabaseResult<()> {
-        for lock in self.changes.iter() {
-            lock.lock().expect("poisoned lock").clear();
-        }
-        self.view_layer.flush()?;
-        self.data_source.flush()
-    }
-}
-
-#[cfg(test)]
-mod tests {
```

Comment:
Do these tests all have parity with new tests? There aren't nearly as many `StorageTransactionInner` tests, but maybe there doesn't need to be.

Reply:
You are right, moved them into `crates/storage/src/transactional.rs`